### ISSUE TITLE
Deprecate and remove go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go-contrib#5551]
This module has been identified to not have an [owner](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/5d8c6d20b1852e47e15b989abe813c20cb935e9b/CODEOWNERS). Based on the [project's ownership policy](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2834#issuecomment-1282698918), this module will be deprecated and then removed.

### How to keep this module

For this module to continue in this repository, it needs a sponsor.

If you would like to sponsor this module and become an owner, please comment in this issue about your desire. As an owner you will assume the following responsibilities:

- You will need to [be a member of the OpenTelemetry organization](https://github.com/open-telemetry/community/blob/main/community-membership.md#member) and maintain that membership.
  - If you are not an existing member, we can add you to the community. This is not a disqualification, but something you will need in the process of becoming a sponsor.
- You will be responsible for keeping up with the instrumented library. Any "upstream" changes that impact this module need to be proactively handle by you.
- You will be expected to review any Pull Requests or Issues created that relate to this module.
- You will be responsible for the stability and versioning compliance of the module.

You will need to have a good working knowledge of the code this module is instrumenting and, ideally, familiarity with the existing module code.

### How this module will be removed

This module is in the process of being deprecated. After that deprecation notice has been published, we will wait 3 months or 2 two releases (whichever is the longer time period). After that time period, this module will be removed from this repository and no more versions of the module will be published.

#### Resurrection

If a sponsor is found after the module has been deprecated or removed, these operations can be reversed (i.e. coded added back, deprecation notice removed).

**Comment by pellared** [comment]
Duplicates https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4678

There is an informal agreement that the MongoDB (Go driver) team will take over the instrumentation library.
Reference: https://jira.mongodb.org/browse/GODRIVER-2938

However, there is no progress so I propose to follow the plan to deprecate it in 4 weeks if no code owner is found.

**Comment by dmathieu** [comment]
I have posted about this removal in the mongo community forum.
https://www.mongodb.com/community/forums/t/otelmongo-at-risk-of-removal/282989

**Comment by pellared** [comment]
I also added a comment in https://jira.mongodb.org/browse/GODRIVER-2938

**Comment by alexbevi** [comment]
@dmathieu / @pellared thank you for reminding us that this still needed an owner. We'll circle back with more details, but apologies for dropping the ball here.

**Comment by ashni-mongodb** [comment]
Hi all, we at MongoDB want to thank the maintainers of the OpenTelemetry Go autoinstrumentation for their hard work and dedication to the community. Please note that we will be assuming ownership of this module, so it will not be deprecated. A few MongoDB folks will plan to attend the next Go SIG in mid-June to discuss further.

If you are using this module and wish to discuss feature requests or ideas for improvement, please reach out via feedback.mongodb.com. Thanks!

**Comment by jtazin** [comment]
Hi @MrAlias, @pellared, and @dmathieu. One of the engineers on our Go driver team, @prestonvasquez, is interested in becoming the code owner for otelmongo. We are currently looking through the steps for becoming an OpenTelemetry member. Also, we will be joining you for the Go SIG meeting on Thursday (June 6).

Thank you once again for the work you’ve done in maintaining the module thus far!

**Comment by prestonvasquez** [comment]
@MrAlias @pellared @dmathieu Thank you for the work you've done maintaining `otelmongo` ❤️  As @jtazin notes, I am interested in becoming the code owner and will work toward membership over the coming week to ensure the continued development of the library.

**Comment by jtazin** [comment]
@pellared @dmathieu Thanks for reviewing and merging [Preston's PR](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5749)!
We would like to continue contributing to otelmongo and have Preston become the official code owner. Would you be willing to sponsor @prestonvasquez in becoming a member of the OpenTelemetry organization?

**Comment by dmathieu** [comment]
The [membership requirements](https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#requirements) state that new members should have multiple contributions to the project.
While Preston's contribution is great, and due to its size, probably counts more than a typo fix, AFAIK, it's the only one. So a membership request seems a bit early.

Also, this is a side note but I'm mentioning it because he's not the one making the request.
Membership and code ownership is an individual thing, not a brand/company thing. That means, if Preston becomes a code owner, that will be under his name, not under the Mongo brand.
Similarly, if he happens to move to another company, code ownership is not interchangeable with another developer (except by contributing and individually becoming an owner).

**Comment by jtazin** [comment]
@dmathieu thanks for your note. your comment regarding membership/ownership is in line with our understanding.

**Comment by prestonvasquez** [comment]
@MrAlias @dmathieu @pellared Here is my activity since the last update:

- https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6171
- https://github.com/open-telemetry/semantic-conventions/issues/1502#issuecomment-2430788127
- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6272
- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6172

Could you give us an update on progress toward sponsorship?

CC: @jtazin

**Comment by pellared** [comment]
First of all, I really want to thank you for your activity.

As for me, given the requirements and responsibilities from both:
- https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#member
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/CONTRIBUTING.md#code-owners

I can sponsor you in both becoming a OTel member and `otelmongo` code owner when you take care of adding `OTEL_SEMCONV_STABILITY_OPT_IN` to `otelmongo` in order to support the latest changes in the semantic conventions. 

Long story short, solving https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6171 is good enough for me.

**Comment by dmathieu** [comment]
I'm also happy to sponsor you once the opt-in environment variable is added to support the latest semconv.

**Comment by prestonvasquez** [comment]
@pellared @dmathieu How do we move forward with sponsorship for org membership and code ownership of otelmongo?

**Comment by prestonvasquez** [comment]
@MrAlias Can this be closed?

**Comment by pellared** [comment]
Closed per https://github.com/open-telemetry/opentelemetry-go-contrib/pull/7242
